### PR TITLE
Exo26

### PR DESCRIPTION
**Pull request by tianyuchen**

### CODE REVIEW

File: 026-Reciprocal_cycles.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    cycle_len = 0
+
+    while quotient not in digits and remainder != 0:
+        digits.append(quotient)
+        quotient, remainder = divmod(10 * remainder, n)
+        print(digits)
+
+    if quotient in digits:
+        digits.pop(0)
+        cycle_len = len(digits)
```
Comment (gameboy1024) [inline 026-Reciprocal_cycles.py]: 这里你写到， digits的长度就是循环的长度
其实不然， 如果现在列表里是：
2，4，6，8，0，6， 那么你又遇到了6， 你第一步pop，ok
然后呢？ 6过了是8，8过了是0， 所以循环的长度应该是从上一次出现的位置（第一个6的下标是多少）到最后的位置（注意你已经把最后一个6去掉了）
你这里写了循环长度=列表总长度，就不对了 而是循环长度等于从重复数字开始到列表末尾的长度。

File: 026-Reciprocal_cycles.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        digits.pop(0)
+        # '- digits.index(quotient)' for the case when repeating after a non-repeating part
+        cycle_len = len(digits) - digits.index(quotient)
+    return cycle_len
+
+maxValue = 0
+for i in range(2, 1000):
+    if recurring_cycle_digits(i) > maxValue:
+        maxValue = recurring_cycle_digits(i)
+        maxIndex = i
```
Comment (gameboy1024) [inline 026-Reciprocal_cycles.py]: maxIndex need to be defined before using

File: 026-Reciprocal_cycles.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        digits.append(quotient)
+        quotient, remainder = divmod((10 ** digitLength) * remainder, n)
+
+    if quotient in digits:
+        digits.pop(0)
+        # '- digits.index(quotient)' for the case when repeating after a non-repeating part
+        cycle_len = len(digits) - digits.index(quotient)
+    return cycle_len
+
+maxValue = 0
```
Comment (gameboy1024) [inline 026-Reciprocal_cycles.py]: Please be consistent with how you name variables.

File: 026-Reciprocal_cycles.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+for i in range(2, 1000):
+    if recurring_cycle_digits(i) > maxValue:
```
Comment (gameboy1024) [inline 026-Reciprocal_cycles.py]: You're unnecessarily calculating recurring_cycle_digits of every number TWICE

File: 026-Reciprocal_cycles.py (hunk 3 above)
Comment (tianyuchen) [reply 026-Reciprocal_cycles.py]: ???

File: 026-Reciprocal_cycles.py (hunk 2 above)
Comment (tianyuchen) [reply 026-Reciprocal_cycles.py]: But there is no error showed...why?

File: 026-Reciprocal_cycles.py (hunk 3 above)
Comment (gameboy1024) [reply 026-Reciprocal_cycles.py]: You're mixing snail_case and camelCase.
Ref:
https://github.com/google/styleguide/blob/gh-pages/pyguide.md#3164-guidelines-derived-from-guidos-recommendations

File: 026-Reciprocal_cycles.py (hunk 2 above)
Comment (gameboy1024) [reply 026-Reciprocal_cycles.py]: Python doesn't require variables to be declared before being used.
You'll have an error if there's a case where this line isn't executed before print(maxIndex):
NameError: name 'maxIndex' is not defined

File: 026-Reciprocal_cycles.py (hunk 3 above)
Comment (gameboy1024) [reply 026-Reciprocal_cycles.py]: camelCase is fine. Do note it's preferred to use snail_case by [Google Python Style Guide](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#3164-guidelines-derived-from-guidos-recommendations).

**Pull request merged**